ert/storage/_storage.py

```diff
@@ -135,46 +131,84 @@ async def _post_to_server_async(
     return resp
 
 
-async def add_record(url: str, record: Any) -> None:
+def _set_content_header(
+    header: str, record_type: ert.data.RecordType, headers: Optional[DictStrAny] = None
+) -> DictStrAny:
+    content_type = _OCTET_STREAM if record_type == ert.data.RecordType.BYTES else _CSV
+    if headers is None:
+        return {header: content_type}
+    headers[header] = content_type
+    return headers
+
+
+async def add_record(url: str, record: ert.data.Record) -> None:
     headers = {
         "Token": StorageInfo.token(),
     }
 
+    assert record.record_type
     if record.record_type != ert.data.RecordType.BYTES:
-        headers["content-type"] = "text/csv"
+        _set_content_header(
+            header="content-type", record_type=record.record_type, headers=headers
+        )
         data = pd.DataFrame([record.data]).to_csv().encode()
         await _post_to_server_async(url=url, headers=headers, data=data)
     else:
         data = {"file": io.BytesIO(record.data)}
         await _post_to_server_async(url=url, headers=headers, files=data)
 
 
-async def load_record(url: str, record_type: Any) -> ert.data.Record:
-    headers = {
-        "Token": StorageInfo.token(),
-    }
+def _interpret_series(row: pd.Series, record_type: ert.data.RecordType) -> Any:
+    if record_type not in {item.value for item in ert.data.RecordType}:
+        raise ValueError(
+            f"Unexpected record type when loading numerical record: {record_type}"
+        )
 
-    if record_type != ert.data.RecordType.BYTES:
-        headers["accept"] = "text/csv"
+    if record_type == ert.data.RecordType.MAPPING_INT_FLOAT:
+        return {int(k): v for k, v in row.to_dict().items()}
+    return (
+        row.to_list()
+        if record_type == ert.data.RecordType.LIST_FLOAT
+        else row.to_dict()
+    )
+
+
+def _response_to_record_collection(
+    content: bytes, metadata: _NumericalMetaData
+) -> ert.data.RecordCollection:
+    record_type = metadata.record_type
+    # is this line really needed? record_type is a RecordType Field
```

Comment:
What line are you referring to? `if record_type not in {item.value for item in ert.data.RecordType}:`?